django/contrib/admin/options.py

```diff
@@ -1194,21 +1185,60 @@ def response_change(self, request, obj):
             self.message_user(request, msg, messages.SUCCESS)
             return self.response_post_save_change(request, obj)
 
+    def admin_viewonsite_url(self, obj):
+        return self.get_view_on_site_url(obj)
+
+    def admin_index_url(self):
+        return self.admin_site.admin_index_url()
+
+    def admin_applist_url(self):
+        return reverse('admin:app_list',
```

Comment:
Use hanging indent as described in [Python coding style](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#python-style) (as you did in `admin_history_url`).